Modules/_threadmodule.c

```diff
@@ -71,6 +75,51 @@ get_thread_state_by_cls(PyTypeObject *cls)
     return get_thread_state(module);
 }
 
+// Helper to set the thread name using platform-specific APIs
+static int
+set_native_thread_name(const char *name)
+{
+#ifdef __APPLE__
+    return pthread_setname_np(name);
+#elif defined(__NetBSD__)
+    pthread_t thread = pthread_self();
+    return pthread_setname_np(thread, "%s", (void *)name);
+#elif defined(HAVE_PTHREAD_SETNAME_NP)
+    pthread_t thread = pthread_self();
+    return pthread_setname_np(thread, name);
+#elif defined(HAVE_PTHREAD_SET_NAME_NP)
+    pthread_t thread = pthread_self();
+    pthread_set_name_np(thread, name);
+    return 0; /* pthread_set_name_np() returns void */
+#else
+    return 0;
+#endif
+}
+
+// Helper to encode and truncate thread name
+static PyObject *
+encode_thread_name(PyObject *name_obj, const char *encoding)
+{
+#ifdef __sun
+    // Solaris always uses UTF-8
+    encoding = "utf-8";
+#endif
+    PyObject *name_encoded = PyUnicode_AsEncodedString(name_obj, encoding, "replace");
+    if (name_encoded == NULL) {
+        return NULL;
+    }
+#ifdef _PYTHREAD_NAME_MAXLEN
+    if (PyBytes_GET_SIZE(name_encoded) > _PYTHREAD_NAME_MAXLEN) {
+        PyObject *truncated = PyBytes_FromStringAndSize(PyBytes_AS_STRING(name_encoded), _PYTHREAD_NAME_MAXLEN);
+        if (truncated == NULL) {
+            Py_DECREF(name_encoded);
+            return NULL;
+        }
+        Py_SETREF(name_encoded, truncated);
+    }
```

Comment:
```suggestion
    if (PyBytes_GET_SIZE(name_encoded) > _PYTHREAD_NAME_MAXLEN) {
        PyObject *truncated = PyBytes_FromStringAndSize(PyBytes_AS_STRING(name_encoded), _PYTHREAD_NAME_MAXLEN);
        Py_DECREF(name_encoded);
        return truncated
    }
```

I think this can be simplified like that.

Reply:
Oh I forgot a semi-colon in my suggestion. You can amend it (after truncated)